src/Cardano/Wallet.hs

```diff
@@ -284,29 +305,39 @@ mkWalletLayer db network = WalletLayer
         withExceptT ErrCreateUnsignedTxCoinSelection $
             CoinSelection.random opts utxo recipients
 
-    , submitTx = \(tx, witnesses) -> do
+    , submitTx = \wid (tx, witnesses) -> do
         let signed = SignedTx $ toByteString $ encodeSignedTx (tx, witnesses)
-        withExceptT NetworkError $ postTx network signed
-
-    , signTx = \wid creds (CoinSelection ins outs chgs) -> DB.withLock db $ do
+        withExceptT ErrSubmitTxNetwork $ postTx network signed
+        let amt = fromIntegral $ sum (getCoin . coin <$> tx ^. #outputs)
```

Comment:
hm, in this case amount will always be equal to sum of all input accounts, as this counts in also change amount. Is it expected from clients that `TxMeta.amount` includes change amount?

Reply:
Ah! Good point. Will correct this.